cabal-install/src/Distribution/Client/SrcDist.hs

```diff
@@ -65,23 +70,45 @@ packageDirToSdist
   -> IO BSL.ByteString
   -- ^ resulting sdist tarball
 packageDirToSdist verbosity gpd dir = do
-  -- let thisDie :: Verbosity -> String -> IO a
-  --    thisDie v s = die' v $ "sdist of " <> prettyShow (packageId gpd) ++ ": " ++ s
-  absDir <- canonicalizePath dir
-  files' <- listPackageSourcesWithDie verbosity dieWithException (Just $ makeSymbolicPath absDir) (flattenPackageDescription gpd) knownSuffixHandlers
-  let files :: [FilePath]
-      files = nub $ sort $ map (normalise . getSymbolicPath) files'
+  let prefix = prettyShow (packageId gpd)
+
+  mbWorkDir <- Just . makeSymbolicPath <$> canonicalizePath dir
+  cabalFilePath <-
+    getSymbolicPath
+      <$> tryFindPackageDesc verbosity mbWorkDir
```

Comment:
You could probably pass in the location, since you must have already read the cabal file, but I do see that `listPackageSources` does also call `tryFindPackageDesc`.